File: app/src/main/java/ru/ls/donkitchen/base/FragmentSubComponent.java
package ru.ls.donkitchen.base;

import com.mobsandgeeks.saripaar.Validator;

import dagger.Subcomponent;
import ru.ls.donkitchen.activity.categorylist.CategoryListFragment;
import ru.ls.donkitchen.activity.receiptdetail.ReceiptDetailFragment;
import ru.ls.donkitchen.activity.receiptdetail.ReceiptDetailInfoFragment;
import ru.ls.donkitchen.activity.receiptdetail.ReceiptDetailReviewsFragment;
import ru.ls.donkitchen.activity.receiptlist.ReceiptListFragment;
import ru.ls.donkitchen.activity.splash.SplashFragment;
import ru.ls.donkitchen.annotation.PerFragment;

@PerFragment
@Subcomponent(modules = FragmentModule.class)
public interface FragmentSubComponent {
	Validator provideValidator();

	void inject(SplashFragment fragment);
	void inject(CategoryListFragment fragment);
	void inject(ReceiptListFragment fragment);
	void inject(ReceiptDetailFragment fragment);
	void inject(ReceiptDetailInfoFragment fragment);
	void inject(ReceiptDetailReviewsFragment fragment);
}

File: app/src/main/kotlin/ru/ls/donkitchen/fragment/base/FragmentSubComponent.kt
package ru.ls.donkitchen.fragment.base

import dagger.Subcomponent
import ru.ls.donkitchen.annotation.PerFragment
import ru.ls.donkitchen.ui.categorylist.CategoryListModule
import ru.ls.donkitchen.ui.categorylist.CategoryListSubComponent
import ru.ls.donkitchen.ui.receiptlist.ReceiptListModule
import ru.ls.donkitchen.ui.receiptlist.ReceiptListSubComponent


/**
 *
 *
 * @author Lord (Kuleshov M.V.)
 * @since 11.01.16
 */
@PerFragment
@Subcomponent(modules = arrayOf(FragmentModule::class))
interface FragmentSubComponent {
//    operator fun plus(module: CategoryListModule): CategoryListSubComponent
//    operator fun plus(module: ReceiptListModule): ReceiptListSubComponent

//    fun inject(fragment: CategoryListFragment)
//    fun inject(fragment: ReceiptListFragment)
//    fun inject(fragment: ReceiptDetailFragment)
//    fun inject(fragment: ReceiptDetailInfoFragment)
//    fun inject(fragment: ReceiptDetailReviewsFragment)
}
